features/integration/applications.feature

```diff
@@ -125,15 +126,79 @@ Feature: Applications
+
+      # create a box called "b64:APj/IA=="
+      And I build an application transaction with the transient account, the current application, suggested params, operation "call", approval-program "", clear-program "", global-bytes 0, global-ints 0, local-bytes 0, local-ints 0, app-args "str:create,b64:APj/IA==", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0, boxes "0,b64:APj/IA=="
+      And I sign and submit the transaction, saving the txid. If there is an error it is "".
+      And I wait for the transaction to be confirmed.
+      Then according to "algod", the contents of the box with name "b64:APj/IA==" in the current application should be "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA". If there is an error it is "".
+
+      # create a box called "b64:MTE0NTE0"
+      And I build an application transaction with the transient account, the current application, suggested params, operation "call", approval-program "", clear-program "", global-bytes 0, global-ints 0, local-bytes 0, local-ints 0, app-args "str:create,b64:MTE0NTE0", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0, boxes "0,b64:MTE0NTE0"
+      And I sign and submit the transaction, saving the txid. If there is an error it is "".
+      And I wait for the transaction to be confirmed.
+      Then according to "algod", the contents of the box with name "b64:MTE0NTE0" in the current application should be "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA". If there is an error it is "".
+
+      # set box "str:foo bar" to value "str:baz qux"
+      And I build an application transaction with the transient account, the current application, suggested params, operation "call", approval-program "", clear-program "", global-bytes 0, global-ints 0, local-bytes 0, local-ints 0, app-args "str:set,str:foo bar,str:baz qux", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0, boxes "0,str:foo bar"
+      And I sign and submit the transaction, saving the txid. If there is an error it is "".
+      And I wait for the transaction to be confirmed.
+      Then according to "algod", the contents of the box with name "str:foo bar" in the current application should be "YmF6IHF1eAAAAAAAAAAAAAAAAAAAAAAA". If there is an error it is "".
+
+      # full check confirmed by both algod and indexer
+      Then according to "algod", the current application should have the following boxes "str:foo bar,b64:APj/IA==,str:name,b64:MTE0NTE0".
+      And according to "algod", by parameter max 2, the current application should have 2 boxes.
+      And according to "algod", by parameter max 6, the current application should have 4 boxes.
+      And I sleep for 1 seconds for indexer to digest things down.
+      And according to "indexer", the current application should have the following boxes "str:foo bar,b64:APj/IA==,str:name,b64:MTE0NTE0".
```

Comment:
I would like to push a little back on the part of `returned in order`, for this would be very strongly related with the order of operations. To couple with parameter `max`/`limit`, one can hardly tell what would be returned under algod.

These are quite implicit and hidden, such that I would not recommend to require these in step for algod.